src/alchemlyb/estimators/ti_.py

```diff
@@ -105,3 +105,50 @@ def fit(self, dHdl):
         self.states_ = means.index.values.tolist()
 
         return self
+
+    def separate_dhdl(self):
+        """
+        For transitions with multiple lambda, the `dhdl` attribute would return
+        a DataFrame which gives the dHdl for all the lambda states, regardless
+        of whether it is perturbed or not. This function creates a list of
+        Series for each lambda, where each Series describes the potential energy
+        gradient for the lambdas state that is perturbed.
+
+        Returns
+        ----------
+        dHdl_list : list
+            A list of Series such that dHdl_list[k] is the potential
+            energy gradient with respect to lambda for each configuration that
+            lambda k is perturbed.
+        """
+        if len(self.dhdl.index.names) == 1:
+            # If only one column is present convert to series
+            if len(self.dhdl.columns) == 1:
+                name = self.dhdl.columns[0]
+                return [self.dhdl[name], ]
+            else: # pragma: no cover
+                raise ValueError('The index column and value column should have'
+                                 ' the same length. Index: {}; Value: {}'
+                                 ''.format(len(self.dhdl.index.names),
+                                           len(self.dhdl.columns)))
```

Comment:
Under which circumstances do you expect this exception to be triggered?

I'd be happier if it gets tested. alchemlyb is trying really hard to have everything tested and experience shows that untested code leads to problems — typically for other people than the author so I'd rather have the author do the work up-front than someone else having to fix it later.

Perhaps you can write a test that provides a very simple dataframe that triggers the exception or you can modify one of the standard test cases in such a way that it fails.

Reply:
I don't think that this will happen in the current setup. I removed this else block.